modules/affile/affile-dest.c

```diff
@@ -479,7 +479,6 @@ affile_dd_reuse_writer(gpointer key, gpointer value, gpointer user_data)
   affile_dw_set_owner(writer, self);
   if (!log_pipe_init(&writer->super))
     {
-      affile_dw_set_owner(writer, NULL);
```

Comment:
I think removing this would cause a leak through the circular reference through the AFFileDestWriter->owner member.

Reply:
I agree, `affile_dw_set_owner()` is necessary here, but its implementation needs to be fixed in order not to dereference NULL pointers. This is a real issue, we've reproduced the crash.

`affile_dw_unset_owner()` would be even cleaner.

Reply:
Thanks once again, `affile_dw_set_owner` got fixed, it should take care of the possible `NULL` owner now.